internal/pkg/manifest/svc.go

```diff
@@ -32,6 +32,30 @@ type ExposedPortsIndex struct {
 	ContainerForPort  map[uint16]string        // holds port to container mapping
 }
 
+func (idx ExposedPortsIndex) findDefaultContainerPort() string {
+	return idx.definedContainerPort(idx.WorkloadName)
+}
+
+// definedContainerPort returns the explicitly defined container port, if there is no port exposed for the container then returns the empty string "".
+func (idx ExposedPortsIndex) definedContainerPort(container string) string {
+	for _, portConfig := range idx.PortsForContainer[container] {
+		if portConfig.isDefinedByContainer {
+			return strconv.Itoa(int(portConfig.Port))
+			/* NOTE: When the `target_port` is empty, the intended target port should be the port that is explicitly exposed by the container. Consider the following example
+			```
+			http:
+			  target_container: nginx
+			sidecars:
+			  nginx:
+			    port: 81 # Explicitly exposed by the nginx container.
+			```
+			In this example, the target port for the ALB listener rule should be 81
+			*/
```

Comment:
nit: this comment is helpful but I feel like it doesn't belong to this function 🤔 The function simply returns the explicitly exposed port from a container, it doesn't care about setting the target port for a listener rule. Perhaps we should move the comment to somewhere relevant to the logic (maybe keep it at the old place).